python/ray/data/tests/test_tf.py

```diff
@@ -28,76 +28,179 @@ def test_autosharding_is_disabled(self):
         expected_auto_shard_policy = tf.data.experimental.AutoShardPolicy.OFF
         assert actual_auto_shard_policy is expected_auto_shard_policy
 
-    def test_element_spec_type(self):
-        ds = ray.data.from_items([{"spam": 0, "ham": 0}])
-
-        dataset = ds.to_tf(feature_columns="spam", label_columns="ham")
+    @pytest.mark.parametrize("include_additional_columns", [False, True])
```

Comment:
Note for reviewers: I intend to validate the API with and without (which shall fall back to the original behavior) the new arguments to ensure that the API is backward compatible. I understand there may be some redundancy in the following tests. Let me know if you think there is a better approach for testing.